java/common/exception/ErrorMessage.java

```diff
@@ -58,17 +58,19 @@
     public static final ErrorMessage INVALID_RULE_WHEN_CONTAINS_DISJUNCTION =
             new ErrorMessage(19, "Rule '%s' 'when' contains a disjunction.");
     public static final ErrorMessage INVALID_RULE_THEN =
-            new ErrorMessage(20, "Rule '%s' 'then' must strictly be either an attribute ownership, or a relation: '%s'.");
+            new ErrorMessage(20, "Rule '%s' 'then' must strictly be a single of either an attribute ownership, or a relation");
+    public static final ErrorMessage INVALID_RULE_THEN_HAS =
+            new ErrorMessage(21, "Rule '%s' 'then' is trying to assign both an attribute type and a variable attribute value: '%s'. Either make the attribute value concrete, or remove the attribute type.");
```

Comment:
so to make this shorter, we could write for instance:
"Rule '%s' cannot assign ownership of a variable attribute '%s' as well as a label '%s'.

I don't exactly like it but I think we can find more succinct language 👍 

who'dve thunk you'd be iterating on sentence structure in a logic-based programming job eeh!?